utils/inference.py

```diff
@@ -58,8 +58,17 @@ def predict_folds(df, train_features, misc_features, fold_id_column, models):
         y_proba = models[0].predict(df[train_features])
         for i in range(y_proba.shape[-1]):
             pred_dict[f'pred_class_{i}_proba'] = y_proba[:,i]
-        pred_dict['pred_class'] = np.argmax(y_proba, axis=-1).astype(np.int32)
-        pred_dict['pred_class_proba'] = np.max(y_proba, axis=-1).astype(np.float32)
+        
+        if cfgparam is not None:
```

Comment:
I propose to extend the labelling with predefined thresholds to a multiclass scenario, since it would be nice to have this functionality for other problems (not necessarily binary). 

I think this can be done firstly by defining a `class_thresholds` dictionary in both `for_evaluation.yaml` and `for_datacards.yaml` configs. Then they are passed to `predict_folds()` as an argument (also checking inside that the thresholds sum to 1) where for each column in `y_proba` a "1" is assigned if the proba passes the corresponding to this class threshold, otherwise 0. Then given this "one hot encoded" representation (checking that there is only the cases that each entry is assigned only to one class), the corresponding class index and probability are stored to `pred_class` and `pred_class_proba` in `pred_dict`. 

But it's just the first idea I thought of how this can be implemented, so if you have a more elegant implementation it is more then welcome🙂